### PR TITLE
UI : Added Flag to exclude bot from user search

### LINKED ISSUE #9497: Do not list bot users in the users' list or remove the delete option for the system's users
**Affected module**
UI/Backend

**Describe the bug**
We are listing bots in the users' list, which can be deleted.  We should:
1. List them but remove the possibility of removing them from UI.
2. Stop listing them.

Before deleting a bot user, we should check if it is associated with a bot.

**To Reproduce**
Search `ingestion-bot` in the list of users.

**Expected behavior**
It should:
1. Gray out the delete button
2. Stop listing them

In any case, do not allow delete a user if it is associated with a bot.

**Version:**
 - OpenMetadata version: 0.13.0

**Additional context**

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes :

- Issue : Closes #9497
- Added Flag to exclude bot from user search

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Frontend Preview (Screenshots) :
Before : 
image

After : 
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Users.spec.js [hunk 1]
```diff
@@ -27,6 +27,26 @@ const userEmail = `${userName}@gmail.com`;
 const adminName = `Admincttest${uuid()}`;
 const adminEmail = `${adminName}@gmail.com`;
 
+const searchBotUser = () => {
+  // Search the created user
+  interceptURL(
+    'GET',
+    '/api/v1/search/query?q=**(isBot:false)&from=0&size=*&index=*',
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Users.spec.js:34]:
```suggestion
    '/api/v1/search/query?q=(isBot:false)&from=0&size=*&index=*',
```

**Comment by Ashish8689** [comment]
@Sachin-chaurasiya  changes updated!

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Users.spec.js [hunk 2]
```diff
@@ -75,6 +101,10 @@ describe('Users flow should work properly', () => {
     softDeleteUser(userName);
     deleteSoftDeletedUser(userName);
   });
+
+  //   it('Search bot user', () => {
+  //     searchBotUser();
+  //   });
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Users.spec.js:107]: Remove this

Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/cypress/e2e/Pages/Users.spec.js]: @Sachin-chaurasiya  removed!

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**